include/boost/geometry/extensions/io/plotly/plotter.hpp

```diff
@@ -0,0 +1,242 @@
+                  << geometry::get<2>(*it)
+                  << "\"";
+            }
+        }
+
+        x << "]";
+        y << "]";
+        z << ( Dimension > 2 ? "]" : "" );
+        
+        os << x.rdbuf()
+           << y.rdbuf() 
+           << z.rdbuf()
+           << ", \"line\": {\"dash\": \"solid\", \"color\": \"" << color << "\", \"width\": 2}";
+    }
+};
+
+
+template <typename MultiGeometry, typename Policy>
+struct json_multi
+{
+    template <typename Char, typename Traits>
+    static inline void apply(std::basic_ostream<Char, Traits>& os,
+        MultiGeometry const& multi, std::string const& color)
+    {
+        for (typename boost::range_iterator<MultiGeometry const>::type
+                    it = boost::begin(multi);
+            it != boost::end(multi);
+            ++it)
+        {
+            Policy::apply(os, *it, color);
+        }
+
+    }
+
+};
+
+}} // namespace detail::json
+#endif // DOXYGEN_NO_DETAIL
+
+#ifndef DOXYGEN_NO_DISPATCH
+namespace dispatch
+{
+
+template 
+<
+    typename Geometry, 
+    typename Tag = typename tag<Geometry>::type
+>
+struct json_plot
+{
+    BOOST_MPL_ASSERT_MSG
+        (
+            false, NOT_OR_NOT_YET_IMPLEMENTED_FOR_THIS_GEOMETRY_TYPE
+            , (Geometry)
+        );
+};
+
+template <typename Point>
+struct json_plot<Point, point_tag> : detail::json::json_point<Point, dimension<Point>::value> {};
+
+template <typename Segment>
+struct json_plot<Segment, segment_tag> : detail::json::json_segment<Segment, dimension<Segment>::value> {};
+
+template <typename Linestring>
+struct json_plot<Linestring, linestring_tag>
+    : detail::json::json_range<Linestring,  dimension<Linestring>::value> {};
+
+template <typename MultiPoint>
+struct json_plot<MultiPoint, multi_point_tag>
+    : detail::json::json_multi
+        <
+            MultiPoint,
+            detail::json::json_point
+                <
+                    typename boost::range_value<MultiPoint>::type,
+                    dimension<MultiPoint>::value
+                >
+        >
+{};
+
+template <typename MultiLinestring>
+struct json_plot<MultiLinestring, multi_linestring_tag>
+    : detail::json::json_multi
+        <
+            MultiLinestring,
+            detail::json::json_range
+                <
+                    typename boost::range_value<MultiLinestring>::type,
+                    dimension<MultiLinestring>::value
+                >
+
+        >
+{};
+
+} // namespace dispatch
+#endif // DOXYGEN_NO_DISPATCH
+
+template <typename Geometry>
+inline void json_plot(std::ostream& stream, Geometry const& geometry, std::string const& color)
+{
+    dispatch::json_plot<Geometry>::apply(stream, geometry, color);
+}
+
+
+class json_plotter : boost::noncopyable
+{
+    std::ostream& m_stream;
+    bool first;
+
+public:
+
+    json_plotter(std::ostream& stream)
+        : m_stream(stream)
+    {
+        m_stream << "{"
+                << std::endl
+                << "    data: [";
```

Comment:
Newlines and indentations are only for human readability. I propose to make them optional. I also propose to replace `std::endl` with `'\n'` because you don't need to flush the stream each time you go to the newline.